dom-event-delegation/main.ts

```diff
@@ -0,0 +1,17 @@
+const $taskList = document.querySelector('.task-list');
+
+if (!$taskList) throw new Error('$taskList query failed');
+
+function handleClick(event: Event): void {
+  const $eventTarget = event.target as HTMLElement;
+  console.log('$eventTarget:', $eventTarget);
+  console.log('$eventTarget.tagName:', $eventTarget.tagName);
+
+  if ($eventTarget.tagName === 'BUTTON') {
+    const $closestListItem = $eventTarget.closest('.task-list-item');
+    console.log('$closestListItem:', $closestListItem);
+    $closestListItem?.remove();
+  }
+}
+
```

Comment:
Another way to do this is by using a "guard clause." Essentially instead of running code `if` something is true, we exit the function early `if` it is false, saving us the time of running through the rest of the function. It also makes things a little easier to read.

```
if ($eventTarget.tagName !== 'BUTTON') {
    return;
  }

  const $taskListItem = $eventTarget.closest('.task-list-item');

  console.log('closest .task-list-item', $taskListItem);
  $taskListItem?.remove();
```